gcloud-java-bigquery/src/main/java/com/google/gcloud/bigquery/BigQuery.java

```diff
@@ -25,5 +34,602 @@
  */
 public interface BigQuery extends Service<BigQueryOptions> {
 
-  // TODO(mziccard) add missing methods
+  enum DatasetField {
+    ACCESS("access"),
+    CREATION_TIME("creationTime"),
+    DATASET_REFERENCE("datasetReference"),
+    DEFAULT_TABLE_EXPIRATION_MS("defaultTableLifetime"),
+    DESCRIPTION("description"),
+    ETAG("etag"),
+    FRIENDLY_NAME("friendlyName"),
+    ID("id"),
+    LAST_MODIFIED_TIME("lastModifiedTime"),
+    LOCATION("location"),
+    SELF_LINK("selfLink");
+
+    private final String selector;
+
+    DatasetField(String selector) {
+      this.selector = selector;
+    }
+
+    public String selector() {
+      return selector;
+    }
+
+    static String selector(DatasetField... fields) {
+      HashSet<String> fieldStrings = Sets.newHashSetWithExpectedSize(fields.length + 1);
```

Comment:
Yes and we should probably reuse that same interface in `Storage` and other services that use field selection. Do you mind if we make this part of a separate PR once bigquery is merged? That would make easier to put the interface in `gcloud-java-core` and modify also `gcloud-java-storage` (no conflicts on merge)